src/Amazon.Lambda.Tools/TemplateProcessor/UpdatableResource.cs

```diff
@@ -140,13 +141,44 @@ public bool IsCode
                     {
                         return false;
                     }
-                    if(!string.IsNullOrEmpty(this._resource.DataSource.GetValue("Code", "ZipFile")))
+                    if (!string.IsNullOrEmpty(this._resource.DataSource.GetValue("Code", "ZipFile")))
                     {
                         // The template contains embedded code.
                         return false;
-                    }                     
+                    }
+                    else
+                    {
+                        string localPath = GetLocalPath();
+
+                        string tempLocalPath = this._resource.DataSource.GetValueFromResource(LambdaConstants.CF_SERVERLESS_METADATA, LambdaConstants.CF_SERVERLESS_DOCKERCONTEXT);
+                        if (string.IsNullOrEmpty(tempLocalPath))
+                        {
+                            tempLocalPath = this._resource.DataSource.GetValue(LambdaConstants.CF_LAMBDA_IMAGEURI);
+                            if (localPath == tempLocalPath && IsECRImage(localPath))
+                            {
+                                return false;
+                            }
+                        }
+
+                        return true;
+                    }
+                }
+            }
 
-                    return true;
+            public bool IsImageUri
+            {
+                get
+                {
+                    string localPath = GetLocalPath();
+
+                    string tempLocalPath = this._resource.DataSource.GetValueFromResource(LambdaConstants.CF_SERVERLESS_METADATA, LambdaConstants.CF_SERVERLESS_DOCKERCONTEXT);
+                    if (string.IsNullOrEmpty(tempLocalPath))
```

Comment:
Same comment as above that the flow and naming is confusing. Maybe you could use the same pattern I suggest above for this section.